common/events/player_event_logs.cpp

```diff
@@ -123,21 +134,466 @@ void PlayerEventLogs::ProcessBatchQueue()
 
 	BenchTimer benchmark;
 
+	EtlQueues etl_queues{};
+
+	for (auto &r:m_record_batch_queue) {
+		switch (r.event_type_id && m_settings[r.event_type_id].etl_enabled) {
+			case PlayerEvent::EventType::LOOT_ITEM: {
+				PlayerEvent::LootItemEvent in{};
+				PlayerEventLootItemsRepository::PlayerEventLootItems out{};
+
+				{
+					std::stringstream ss;
+					ss << r.event_data;
+					cereal::JSONInputArchive ar(ss);
+					in.serialize(ar);
+				}
+
+				out.charges     = in.charges;
+				out.corpse_name = in.corpse_name;
+				out.item_id     = in.item_id;
+				out.item_name   = in.item_name;
+				out.npc_id      = in.npc_id;
+				out.created_at  = r.created_at;
+
+				if (m_etl_settings.contains(PlayerEvent::EventType::LOOT_ITEM)) {
+					r.etl_table_id = m_etl_settings.at(PlayerEvent::EventType::LOOT_ITEM).next_id;
+					m_etl_settings.at(PlayerEvent::EventType::LOOT_ITEM).next_id++;
+				}
+
+				etl_queues.loot_items.push_back(out);
+				break;
+			}
+			case PlayerEvent::EventType::MERCHANT_SELL: {
+				PlayerEvent::MerchantSellEvent in{};
+				PlayerEventMerchantSellRepository::PlayerEventMerchantSell out{};
+
+				{
+					std::stringstream ss;
+					ss << r.event_data;
+					cereal::JSONInputArchive ar(ss);
+					in.serialize(ar);
+				}
+
+				out.npc_id                  = in.npc_id;
+				out.merchant_name           = in.merchant_name;
+				out.merchant_type           = in.merchant_type;
+				out.item_id                 = in.item_id;
+				out.item_name               = in.item_name;
+				out.charges                 = in.charges;
+				out.cost                    = in.cost;
+				out.alternate_currency_id   = in.alternate_currency_id;
+				out.player_money_balance    = in.player_money_balance;
+				out.player_currency_balance = in.player_currency_balance;
+				out.created_at              = r.created_at;
+
+				if (m_etl_settings.contains(PlayerEvent::EventType::MERCHANT_SELL)) {
+					r.etl_table_id = m_etl_settings.at(PlayerEvent::EventType::MERCHANT_SELL).next_id;
+					m_etl_settings.at(PlayerEvent::EventType::MERCHANT_SELL).next_id++;
+				}
+
+				etl_queues.merchant_sell.push_back(out);
```

Comment:
`.reserve()` to pre-allocate memory since we know the size ahead of time

Reply:
Added reserve to queue.